flink-runtime/pom.xml

```diff
@@ -427,17 +427,42 @@ under the License.
 							<goal>shade</goal>
 						</goals>
 						<configuration>
+							<!-- we need this to avoid having to specify all of akka's dependencies -->
+							<promoteTransitiveDependencies>true</promoteTransitiveDependencies>
```

Comment:
Why exactly do we need this? AFAIK you can shade transitive dependencies without promoting them.

Reply:
This adds `akka-remote`'s dependencies as direct dependencies to `flink-runtime` (like for example `reactive-streams`).
Without promoting transitive dependencies, we would need to add all transitive dependencies of `akka-remote` to the shaded jar, because `akka-remote` is removed as a dependency which also removes its transitive dependencies.

We could do that, but since we cannot relocate many of them (Scala) I though to keep the set of dependencies that are "present but undeclared" small and treat akka-remote's dependencies like regular dependencies.

Reply:
We could also add all transitive dependencies of `akka-remote` to the shaded jar. Its a similar consideration as "add all akka, or only akka-remote".